api/enki.py

```diff
@@ -0,0 +1,633 @@
+                    identifier, "Book not in collection", data_source=self.output_source, transient=False
+                )
+                batch_results.append(result)
+        return batch_results
+
+    def handle_success(self, identifier):
+        return self.set_presentation_ready(identifier)
+
+    def process_item(self, identifier):
+        results = self.process_batch([identifier])
+        return results[0]
+
+class BibliographicParser(object):
+
+    """Helper function to parse JSON"""
+    def process_all(self, json_data, xpath, namespaces=None, handler=None, parser=None):
+        data = json.loads(json_data)
+	returned_titles = data["result"]["titles"]
+	titles = returned_titles
+	for book in returned_titles:
+            print "A book titled '%s'" % book["title"]
+            print book
+            print "\n"
+	    data = self.process_one(book, namespaces)
+            if data:
+                yield data
+
+    DELIVERY_DATA_FOR_AXIS_FORMAT = {
+        "Blio" : None,
+        "Acoustik" : None,
+        "ePub" : (Representation.EPUB_MEDIA_TYPE, DeliveryMechanism.ADOBE_DRM),
+        "PDF" : (Representation.PDF_MEDIA_TYPE, DeliveryMechanism.ADOBE_DRM),
+    }
+
+    log = logging.getLogger("Enki Bibliographic Parser")
+
+    @classmethod
+    def parse_list(self, l):
+        """Turn strings like this into lists:
+
+        FICTION / Thrillers; FICTION / Suspense; FICTION / General
+        Ursu, Anne ; Fortune, Eric (ILT)
+        """
+        return [x.strip() for x in l.split(";")]
+
+    def __init__(self, include_availability=True, include_bibliographic=True):
+        self.include_availability = include_availability
+        self.include_bibliographic = include_bibliographic
+
+    def extract_availability(self, circulation_data, element, ns):
+	primary_identifier = IdentifierData(Identifier.ENKI_ID, element["id"])
+        if not circulation_data:
+            circulation_data = CirculationData(
+                data_source=DataSource.ENKI,
+                primary_identifier=primary_identifier,
+            )
+        # For now, assume there is a license available for each item.
+        circulation_data.licenses_owned=1
+        circulation_data.licenses_available=1
+        circulation_data.licenses_reserved=0
+        circulation_data.patrons_in_hold_queue=0
+
+        return circulation_data
+
+
+    # Axis authors with a special role have an abbreviation after their names,
+    # e.g. "San Ruby (FRW)"
+    role_abbreviation = re.compile("\(([A-Z][A-Z][A-Z])\)$")
+    generic_author = object()
+    role_abbreviation_to_role = dict(
+        INT=Contributor.INTRODUCTION_ROLE,
+        EDT=Contributor.EDITOR_ROLE,
+        PHT=Contributor.PHOTOGRAPHER_ROLE,
+        ILT=Contributor.ILLUSTRATOR_ROLE,
+        TRN=Contributor.TRANSLATOR_ROLE,
+        FRW=Contributor.FOREWORD_ROLE,
+        ADP=generic_author, # Author of adaptation
+        COR=generic_author, # Corporate author
+    )
```

Comment:
Does ENKI happen use the same format for describing contributor roles as Axis does, or is this an artifact of copy-and-paste?